frontend/src/app/app-events.service.ts

```diff
@@ -63,6 +132,23 @@ export class AppEventsService implements OnDestroy {
     } else {
       this.events.next({type: AppEventType.MULTIPLE_PEOPLE});
     }
+
+    var counter = 0;
+
+    for (let i = 0; i < sumPplWave.length; i ++) {
+      if (sumPplWave[i].rwx < sumPplWave[i].rsx && sumPplWave[i].rwy < sumPplWave[i].rsy) {
+        this.events.next({type: AppEventType.ONE_WAVE_DETECTED});
+        counter ++;
+      } else if (sumPplWave[i].lwx > sumPplWave[i].lsx && sumPplWave[i].lwy < sumPplWave[i].lsy) {
+        this.events.next({type: AppEventType.ONE_WAVE_DETECTED});
+        counter ++;
+      }
+    }
+
+    if (counter > 1) {
+      this.events.next({type: AppEventType.TWO_WAVES_DETECTED});
+    }
```

Comment:
For an individual detection pass, let's emit only one of either `ONE_WAVE_DETECTED` or `TWO_WAVES_DETECTED` rather than emitting multiple events per pass when more than one wave is detected. This determination is specific to our application's logic. I would think moving the `next` event out to the if/else on counter below is the right refactoring.